lib/cartopy/io/img_tiles.py

```diff
@@ -350,23 +355,33 @@ class Stamen(GoogleWTS):
     """
 
     def __init__(self, style='toner',
-                 desired_tile_form=None, cache=False):
-
+                 desired_tile_form=None, cache=False,
+                 apikey=None):
+        if apikey is None:
+            raise ValueError("An apikey is now required to use Stamen maps. "
+                             "Visit https://maps.stadiamaps.com/ for more details.")
         # preset layer configuration
         layer_config = {
-          'terrain':            {'extension': 'png', 'opaque': True},
-          'terrain-background': {'extension': 'png', 'opaque': True},
-          'terrain-labels':     {'extension': 'png', 'opaque': False},
-          'terrain-lines':      {'extension': 'png', 'opaque': False},
-          'toner-background':   {'extension': 'png', 'opaque': True},
-          'toner':              {'extension': 'png', 'opaque': True},
-          'toner-hybrid':       {'extension': 'png', 'opaque': False},
-          'toner-labels':       {'extension': 'png', 'opaque': False},
-          'toner-lines':        {'extension': 'png', 'opaque': False},
-          'toner-lite':         {'extension': 'png', 'opaque': True},
-          'watercolor':         {'extension': 'jpg', 'opaque': True},
+          'stamen_terrain':            {'extension': 'png', 'opaque': True},
+          'stamen_terrain_background': {'extension': 'png', 'opaque': True},
+          'stamen_terrain_labels':     {'extension': 'png', 'opaque': False},
+          'stamen_terrain_lines':      {'extension': 'png', 'opaque': False},
+          'stamen_toner_background':   {'extension': 'png', 'opaque': True},
+          'stamen_toner':              {'extension': 'png', 'opaque': True},
+          'stamen_toner_hybrid':       {'extension': 'png', 'opaque': False},
+          'stamen_toner_labels':       {'extension': 'png', 'opaque': False},
+          'stamen_toner_lines':        {'extension': 'png', 'opaque': False},
+          'stamen_toner_lite':         {'extension': 'png', 'opaque': True},
+          'stamen_watercolor':         {'extension': 'jpg', 'opaque': True},
         }
 
+        # Stadia Maps has a different naming convention for the styles
+        # they prepend stamen_ and dashes are now underscores
+        if "stamen" not in style:
+            style = f"stamen_{style.replace('-', '_')}"
+            warnings.warn("Stamen maps now require the stamen_ prefix for "
+                          f"the style name, please update to the new style: {style}")
+
```

Comment:
This gets to your question about potentially switching the class name over to Stadia, but the name replacement here does not seem ideal since it would prevent using styles now available like Alidade Smooth

Reply:
Yes, that seems preferable to me. Since the user has to get an api key from Stadia, might as well make all the styles available!

Reply:
Yeah, this is the approach taken by OpenLayers, Leaflet Providers, and more. It's already a breaking change, so changing the name makes sense.

You can find a full list of our styles here: https://docs.stadiamaps.com/themes/. Some notes:

* Stamen styles are (currently) the only ones with multiple variations per style. All others have just one identifier.
* Alidade Satellite is _not_ currently available as pure raster tiles; we expect that to change in the future, but probably doesn't make sense to include that one now.